### ISSUE TITLE
tunneling socket could not be established, statusCode=407

### ISSUE THREAD

**Issue by coreywebber** [actions/runner#880]
**Describe the bug**
So, I know the error is related to a proxy authorization, but the bug is more why it's happening now. For months I have been running the same YAML, with no issues. Now when I run them, I get this error. I have searched the logs and I am not seeing anything that stands out. I have also tried my personal credentials as the service account and at the proxy, all yielding the same results. There could be other reasons I am sure, ex. Windows Updates, but nothing else is experiencing this issue.

**To Reproduce**
When the action starts, 
Step 1
- it connects to GitHub to download the code
- it compiles the code via msbuild
- uploads artifacts
It does this without any issue. 

Step 2
- Download artifacts.
   - Here I receive a 407 error
- Deploy to Test (Skipped)
- Deploy to Prod (Skipped)

**Expected behavior**
To pass the proxy authentication to all actions

## Runner Version and Platform
Version of your runner? 2.275.1

OS of the machine running the runner? Windows

## What's not working?
```
Run actions/download-artifact@v2
  with:
    name: warehouse-artifact
Starting download for warehouse-artifact
(node:4716) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
Error: tunneling socket could not be established, statusCode=407
```
## Job Log Output
```
2020-12-24T04:02:26.7309148Z ##[section]Starting: Request a runner to run this job
2020-12-24T04:02:27.2484788Z Can't find any online and idle self-hosted runner in current repository that matches the required labels: 'bdw'
2020-12-24T04:02:27.2484912Z Found online and idle self-hosted runner in current repository's account/organization that matches the required labels: 'bdw'
2020-12-24T04:02:27.3467951Z ##[section]Finishing: Request a runner to run this job
2020-12-24T04:02:31.0630378Z Current runner version: '2.275.1'
2020-12-24T04:02:31.0641677Z Runner name: 'BizOps'
2020-12-24T04:02:31.0642374Z Runner group name: 'BizOps - Data Warehouse'
2020-12-24T04:02:31.0643997Z Machine name: 'AUSPWBOOMIATOM4'
2020-12-24T04:02:31.0651204Z Runner is running behind proxy server '***proxy:80' for all HTTP requests.
2020-12-24T04:02:31.0652634Z Runner is running behind proxy server '***proxy:80' for all HTTPS requests.
2020-12-24T04:02:31.0653362Z Prepare workflow directory
2020-12-24T04:02:31.1486168Z Prepare all required actions
2020-12-24T04:02:31.2014545Z Getting action download info
2020-12-24T04:02:31.5119967Z Download action repository 'actions/download-artifact@v2'
2020-12-24T04:02:32.5511684Z ##[group]Run actions/download-artifact@v2
2020-12-24T04:02:32.5512515Z with:
2020-12-24T04:02:32.5513232Z   name: warehouse-artifact
2020-12-24T04:02:32.5514118Z ##[endgroup]
2020-12-24T04:02:32.6880549Z Starting download for warehouse-artifact
2020-12-24T04:02:32.7053235Z (node:4716) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
2020-12-24T04:02:32.7307715Z ##[error]tunneling socket could not be established, statusCode=407
2020-12-24T04:02:32.7669558Z Cleaning up orphan processes

```

## Runner and Worker's Diagnostic Logs
[Runner_20201224-035016-utc.log](https://github.com/actions/runner/files/5738134/Runner_20201224-035016-utc.log)
[Worker_20201224-035100-utc.log](https://github.com/actions/runner/files/5738146/Worker_20201224-035100-utc.log)
[Worker_20201224-040229-utc.log](https://github.com/actions/runner/files/5738135/Worker_20201224-040229-utc.log)

**Comment by okainov** [comment]
Why was it closed? What is the solution?

**Comment by coreywebber** [comment]
I stopped using download-artifact to be honest, but one of the issues that I came across was that the GIT proxy was not set in the bash in environment (in windows). 

This ended up fixing most issues (outside this one) but the download-artifact never worked. 
I tried:
- .env file (which seems to only effect the runner)
- Environment Variables ( didnt see to have an effect)
- Investigated the issue is in JavaScript behind the task, but nothing stood out.

So last year I pressed on. I closed this out with a few others, and can revive it, cause I still have the issue.

**Comment by coreywebber** [comment]
I stopped using download-artifact to be honest, but one of the issues that I came across was that the GIT proxy was not set in the bash in environment (in windows). 

This ended up fixing most issues (outside this one), but the download-artifact never worked. 
I tried:
- .env file (which seems to only effect the runner)
- Environment Variables (didn't seem to have an effect)
- Investigated the issue is in JavaScript behind the task, but nothing stood out.

So last year I pressed on. I closed this out with a few others, and can revive it, because I still have the issue.
